STATIC PREDICTION TABLES.md

```diff
@@ -0,0 +1,96 @@
+# About this document
+
+This is a Work in Progress.
+
+This document describes a format for storing Static Prediction Tables.
+
+It is possible that some of all of this format will be usable as part of
+individual compressed source files, to be determined.
+
+# About Static Prediction Tables (SPT)
+
+A Static Prediction Table is a form of external dictionary, shipped either
+with the JavaScript VM, or separately, and which may be referenced by any
+number of compressed Binary AST Source Files.
+
+Shipping Static Prediction Tables makes it possible to considerably reduce
+the size of individual compressed files.
+
+This document does not attempt to document how and when a SPT is loaded,
+or how and when an individual compressed file references a SPT.
+
+# Design guidelines
+
+- A SPT must be usable by many compressed source files.
+- A VM must be able to manage several SPTs simultaneously.
+- As JavaScript is a changing language, a SPT that is complete at a given point in time may not be expected to remain
+  complete forever.
+- As Binary AST never reuses the same interface name for distinct purposes, a Path in the AST that is valid at a point
+  in time will remain valid forever.
+- For upgrade purposes, a SPT may be defined as an amendment to another SPT.
+- A SPT may define additional strings of various natures.
+
+# Format
+
+## Header
+
+The header:
+- specifies the kind of file;
+- references the grammar version;
+- optionally, references a SPT file it amends.
+
+TBD
+
+## Tables of Strings
+
+These tables add new strings that may be referenced both in the tables of probabilities
```

Comment:
So far, the probability tables for strings just predict indexes into a move-to-front cache.  We do not actually need to assign general probabilities to the string table itself - they will be predicted well after they are first referenced (and encoded using some varuint-encoding), and subsequently added to the MoveToFront String cache.

Reply:
That's true for string literals, identifier names and property keys.

On the other hand, it's not true for interface names and string enums.

I'll amend the text to clarify.